### PR TITLE
Add NPM detection of sensitive data exfiltration javascript code 

### PR DESCRIPTION
**Pull request by sobregosodd**
Adds new rule that detects javascript code being purposefully obfuscated.
The goal is to detect the most common patterns used to exfiltrate data such as:

* Usual sensitive files being read
* Usual outbound http 
* Usual exfiltration using other than http libraries

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,6 +105,7 @@ Metadata heuristics:
 | release_zero | Identify packages with an release version that's 0.0 or 0.0.0 |
 | typosquatting | Identify packages that are named closely to an highly popular package |
 | potentially_compromised_email_domain | Identify when a package maintainer e-mail domain (and therefore package manager account) might have been compromised |
+| unclaimed_maintainer_email_domain | Identify when a package maintainer e-mail domain (and therefore package manager account) is currently unclaimed and could be registered by an attacker |
```
Comment (christophetd) [inline README.md:108]:
```suggestion
| unclaimed_maintainer_email_domain | Identify when a package maintainer e-mail domain (and therefore npm account) is unclaimed and can be registered by an attacker |
```

File: README.md [hunk 2]
```diff
@@ -121,14 +122,17 @@ Source code heuristics:
 | shady-links | Identify when a package contains an URL to a domain with a suspicious extension |
 | npm-exec-base64 | Identify when a package dynamically executes code through 'eval' |
 | npm-install-script | Identify when a package has a pre or post-install script automatically running commands |
+| npm-exfiltrate-sensitive-data | Identify when a package reads and exfiltrates sensitive data from the local system |
+| npm-exfiltrate-sensitive-data | Identify when a package reads and exfiltrates sensitive data from the local system |
```
Comment (christophetd) [inline README.md:126]: duplicate?

Comment (sobregosodd) [reply README.md:126]: It seems a bug with the doc generator when a rule has two languages belonging to the same ecosystem. e.g: javascript, typescript
I'll fix it

Comment (sobregosodd) [reply README.md]: fixed in https://github.com/DataDog/guarddog/pull/346/commits/d10a646a725f3ccb6d66a5c9e5dce7326e1b542d

File: README.md (hunk 1 above)
Comment (sobregosodd) [reply README.md]: fixed in https://github.com/DataDog/guarddog/pull/346/commits/d10a646a725f3ccb6d66a5c9e5dce7326e1b542d

**Review by christophetd — APPROVED** [review]

**Pull request merged**